src/Api.ts

```diff
@@ -1,27 +1,22 @@
 import { EmbalmerFacet__factory } from '@sarcophagus-org/sarcophagus-v2-contracts';
-import { BigNumber, ethers } from 'ethers';
-import { SarcoClient } from './SarcoClient';
+import { ethers } from 'ethers';
 import { safeContractCall } from './helpers/safeContractCall';
 import { CallOptions } from './types';
-import { goerliNetworkConfig } from './networkConfig';
 import {
   SarcophagusSettings,
   sarcophagusSettingsSchema,
   archaeologistSettingsArraySchema,
   ArchaeologistSettings,
 } from './helpers/validation';
+import { getSarcophagusRewraps } from './helpers/subgraph';
 
 export class Api {
-  sarcoClient: SarcoClient;
-  embalmerFacet: ethers.Contract;
+  private embalmerFacet: ethers.Contract;
+  private subgraphUrl: string;
 
-  constructor(sarcoClient: SarcoClient) {
-    this.sarcoClient = sarcoClient;
-    this.embalmerFacet = new ethers.Contract(
-      goerliNetworkConfig.diamondDeployAddress,
-      EmbalmerFacet__factory.abi,
-      this.sarcoClient.signer
-    );
+  constructor(diamondDeployAddress: string, signer: ethers.Signer, subgraphUrl: string) {
```

Comment:
These constructors have been updated to accept only the parameters they'll need internally, since some SarcoClient methods have been made private and will not be accessible from here